### PR TITLE
Update dependencies and fix error handling

### PR DESCRIPTION
**Pull request by floatdrop**
`read-all-stream@~1` using `on('data')` handler, which forces stream to fall into Stream1 mode and flush all data.

This should be related to #67 (but will not fix it yet).

### CODE REVIEW

File: test/test.js [hunk 1]
```diff
@@ -14,7 +14,7 @@ test('expose a constructor', function (t) {
 
 test('return an instance if it called without `new`', function (t) {
 	t.plan(1);
-	t.assert(Download() instanceof Download);
+	t.assert(new Download() instanceof Download);
```
Comment (shinnn) [inline test/test.js]: Check the test spec.

> 'return an instance if it called without `new`'

Comment (floatdrop) [inline test/test.js]: :+1:

**Comment by shinnn** [comment]
LGTM

**Comment by kevva** [comment]
Thank you :).

**Comment by kevva** [comment]
Also, published to npm!

**Pull request merged**